packages/core/src/application.ts

```diff
@@ -312,6 +314,23 @@ export class Application extends Context implements LifeCycleObserver {
     this.setState('started');
   }
 
+  /**
+   * Register a function to be called when the application starts.
+   *
+   * This is a shortcut for adding a binding for a LifeCycleObserver
+   * implementing a `start()` method.
+   *
+   * @param fn The function to invoke, it can be synchronous (returning `void`)
+   * or asynchronous (returning `Promise<void>`).
+   * @returns The LifeCycleObserver binding created.
+   */
+  public onStart(fn: () => ValueOrPromise<void>): Binding<LifeCycleObserver> {
```

Comment:
Do we want to allow an optional `group` as the 2nd argument for ordering the life cycle observer?

Reply:
It makes sense to me to eventually support `group` argument, but since I don't need it myself, I prefer to leave that out of my pull request. [YAGNI](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it).

Reply:
It would be nice to support it as part of this PR to make the feature complete without requiring us to revisit it. But feel free to defer it.

Reply:
> It would be nice to support it as part of this PR to make the feature complete without requiring us to revisit it.

I know! On the other hand, if we leave `group` out and somebody comes back to use asking for that feature, then we will validate usefulness of this feature and provide a good opportunity for the reporter to contribute the fix, growing our community of code contributors by one more person.